File: kotlin/3. array/lecture 1/max consecutive ones.kt
class Solution {
    // In Kotlin, IntArray - primitive arrays (like int[] in Java)
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var maxCount = 0
        var count = 0

        for (num in nums) {
            if (num == 1) {
                count++
                maxCount = maxOf(maxCount, count)
            } else {
                count = 0
            }
        }

        return maxCount
    }
}

fun main() {
    val sol = Solution()
    val nums = intArrayOf(1, 1, 0, 1, 1, 1)
    
    println("Maximum consecutive ones: ${sol.findMaxConsecutiveOnes(nums)}")
}

File: 2. python/3. array-list/lecture 1/max consecutive ones.py
from typing import List

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_count = 0
        count = 0

        for num in nums:
            if num == 1:
                count += 1
                max_count = max(max_count, count)
            else:
                count = 0
        
        return max_count

# This block acts as 'int main()'
if __name__ == "__main__":
    sol = Solution()
    nums = [1, 1, 0, 1, 1, 1]
    
    print(f"Maximum consecutive ones: {sol.findMaxConsecutiveOnes(nums)}")